### PR TITLE
Restructure scheduler storage module

### PR DESCRIPTION
**Pull request by jpbruinsslot**
### Changes

Minor changes to the storage module.

* method name `get_task` change to fall in line with the other datastores and method naming
* update  `DBConn.__init__` to use an attribute logging module instead of a 'global' one.

### QA notes

Changes done to the datastores, the main QA points of attention would be that the scheduler would perform its normal procedures.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by jpbruinsslot** [comment]
Fixing mypy errors atm

File: mula/scheduler/storage/storage.py [hunk 1]
```diff
@@ -1,68 +1,67 @@
...
 from scheduler.config import settings
 
-logger = logging.getLogger(__name__)
+from .errors import StorageError
 
 
 class DBConn:
-    def __init__(self, dsn: str) -> None:
-        super().__init__()
+    def __init__(self, dsn: str, pool_size: int = 25) -> None:
```
Comment (ammar92) [inline mula/scheduler/storage/storage.py:13]: Although the initializer mustn't return anything (except for `None`), it does not need to and should not have a return type

File: mula/scheduler/storage/storage.py [hunk 2]
```diff
@@ -1,68 +1,67 @@
...
+        self.logger: structlog.BoundLogger = structlog.getLogger(__name__)
 
-        self.engine = None
+        self.dsn: str = dsn
+        self.pool_size: int = pool_size
```
Comment (ammar92) [inline mula/scheduler/storage/storage.py:17]: The type hints here are redundant, they can be inferred (by users and type checkers) from type hints given in the signature

**Review by ammar92 — COMMENTED** [review]
Looks good to me 👍 I just have a few suggestions and notes to improve it even more

File: mula/scheduler/storage/pq_store.py [hunk 3]
```diff
@@ -2,8 +2,10 @@
 
 from scheduler import models
 
+from .errors import exception_handler
```
[the remark this answers is not in the document]
Comment (jpbruinsslot) [reply mula/scheduler/storage/pq_store.py:5]: Yeah I agree, has been used inconsistently.

File: mula/scheduler/context/context.py [hunk 4]
```diff
@@ -149,8 +151,18 @@ def __init__(self) -> None:
             }
         )
 
+        # Database connection
+        try:
+            dbconn = storage.DBConn(str(self.config.db_uri))
+            dbconn.connect()
+        except storage.errors.StorageError as e:
+            self.logger.error("Failed to connect to database", exc_info=e)
```
[the remark this answers is not in the document]
Comment (jpbruinsslot) [reply mula/scheduler/context/context.py:159]: Let's see if can remember that for the future 🙈

File: mula/scheduler/storage/pq_store.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpbruinsslot) [reply mula/scheduler/storage/pq_store.py:5]: Indeed let's discuss this and find a good solution. Figured I just try it, but hit circular dependency issue with it. And it'll probably be good to implement this for the whole scheduler application.

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Added 2 organisations and performed scanning activities with both. The tasks seem to run and complete and corresponding findings are created. 

Ready for merge. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**